File: backend/src/main/java/com/ubre/backend/dto/NotificationDto.java
package com.ubre.backend.dto;

import com.ubre.backend.enums.NotificationType;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.validator.constraints.Normalized;

// For system notifications

@Getter
@Setter
@NoArgsConstructor
public class NotificationDto {
    private Long id;
    private Long userId;
    private String title;
    private String message;
    private Boolean read;
    private NotificationType type;

    public NotificationDto(Long id, Long userId, String title, String message, boolean read, NotificationType type) {
        this.id = id;
        this.userId = userId;
        this.title = title;
        this.message = message;
        this.read = read;
        this.type = type;
    }
}

File: frontend/ubre/src/app/dtos/notification-dto.ts
import { NotificationType } from "../enums/notification-type";

export class NotificationDto {
    public id : number;
    public userId : number;
    public title : String;
    public message : String;
    public read : boolean;
    public type : NotificationType;

    constructor(id : number, userId : number, title : String, message : String, read : boolean, type : NotificationType) {
        this.id = id;
        this.userId = userId;
        this.title = title;
        this.message = message;
        this.read = read;
        this.type = type;
    }
}
